### ISSUE TITLE
Sandbox security problem?

### ISSUE THREAD

**Issue by dabrahams** [apple/swift-package-manager#6951]
### Description

I was surprised to find that if I make an outer SPM invocation, and within that, a nested SPM invocation with --disable-sandbox that I can write on directories outside of the outer sandbox.  Is that intentional? It seems like a bit of a security problem that I can bypass a sandbox just by invoking spm from within it with the right options.

### Swift Package Manager version/commit hash

Swift 5.9

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
Target: arm64-apple-macosx13.0
Darwin davea-mbp16.corp.adobe.com 22.6.0 Darwin Kernel Version 22.6.0: Fri Sep 15 13:41:28 PDT 2023; root:xnu-8796.141.3.700.8~1/RELEASE_ARM64_T6000 arm64

**Comment by neonichu** [comment]
What do you mean by "within that"? You're executing another SwiftPM via a build or command plugin?

**Comment by dabrahams** [comment]
Yes via build tool plugin

**Comment by neonichu** [comment]
And that's invoking SwiftPM from the plugin directly or producing a build command instance that will use SwiftPM?

**Comment by dabrahams** [comment]
What I did was to take the portable-build-tool branch of [this project](https://github.com/dabrahams/TestGeneration/blob/portable-build-tool/Plugins/ResourceGeneratorPlugin/PortableBuildToolPlugin.swift) and made most of the `#if os(Windows)` code in the linked file (except for the part that fixes paths) run unconditionally.  I don't have the patch handy, but it's mostly a matter of redefining those constants declared near the top.  The build tool invokes `swift run` as you can see.  That I was able to `swift test` on  Mac without `--disable-sandbox` surprised me.  The fact that the inner `swift run` has the sandbox disabled should be irrelevant to whether you're allowed to make a nested sandbox inside the outer one… right?

**Comment by neonichu** [comment]
Right, so it looks like you are producing build commands with SwiftPM. I wonder if those are potentially not sandboxed?

**Comment by neonichu** [comment]
> Right, so it looks like you are producing build commands that execute SwiftPM. I wonder if those are potentially not sandboxed?

No, that is not the case, at least in theory, we are sandboxing both: https://github.com/apple/swift-package-manager/blob/main/Sources/Build/LLBuildManifestBuilder.swift#L230
